pulp_container/app/tasks/builder.py

```diff
@@ -0,0 +1,133 @@
+from pulp_container.app.models import (
+    Blob,
+    BlobManifest,
+    ContainerRepository,
+    Manifest,
+    Tag
+)
+from pulp_container.constants import MEDIA_TYPE
+from pulpcore.app.models.content import Artifact, ContentArtifact
+from pulpcore.plugin.tasking import WorkingDirectory
+
+import json
+import os
+import shutil
+import subprocess
+from uuid import uuid4
+
+
+def get_or_create_blob(layer_json, manifest, path):
+    """
+    Creates Blob from json snippet of manifest.json
+
+    Args:
+        layer_json (json): json
```

Comment:
we need to distinguish between a blob and config blob and make different relations, otherwise we'll endup having wrong relations where config blob will be present among blobs=[] and config_blog is null
```
$ http GET :24817/pulp/api/v3/content/container/manifests/?repository_version=/pulp/api/v3/repositories/container/container/4512857f-b88a-4974-88c7-17f33e8676ec/versions/2/
HTTP/1.1 200 OK
Allow: GET, HEAD, OPTIONS
Connection: close
Content-Length: 1363
Content-Type: application/json
Date: Wed, 15 Jan 2020 12:23:07 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "artifact": "/pulp/api/v3/artifacts/fc7dfadf-ecf0-4aa6-8bb8-6087d12ddc6e/",
            "blobs": [
                "/pulp/api/v3/content/container/blobs/003e30fe-9196-4014-8b67-21d8301c8338/",
                "/pulp/api/v3/content/container/blobs/3b67f8d4-65f4-4b87-8194-a409788bd486/",
                "/pulp/api/v3/content/container/blobs/ceaa3fd3-7479-401a-88c9-77dc1dc15f21/"
            ],
            "config_blob": null,
            "digest": "sha256:92584bd3bea296ae4eb862dfbf7ab069d06f5f1af0cd9a11ba5d5f973b041c77",
            "listed_manifests": [],
            "media_type": "application/vnd.oci.image.manifest.v1+json",
            "pulp_created": "2020-01-15T12:22:38.982892Z",
            "pulp_href": "/pulp/api/v3/content/container/manifests/a93c3eb1-d029-427a-b306-00144cc7aa0e/",
            "schema_version": 2
        },
        {
            "artifact": "/pulp/api/v3/artifacts/c524c6af-3835-45e6-91de-e1168dacc849/",
            "blobs": [
                "/pulp/api/v3/content/container/blobs/5f7ed8c6-8834-4407-83f1-1da5cf894562/",
                "/pulp/api/v3/content/container/blobs/003e30fe-9196-4014-8b67-21d8301c8338/",
                "/pulp/api/v3/content/container/blobs/82661ee7-1a5e-40aa-972b-4b755127f372/"
            ],
            "config_blob": null,
            "digest": "sha256:96af072fd0e111b600a2ecf9ab2da017fc8d1c3bd3ffeabd1639bd81f75d18e7",
            "listed_manifests": [],
            "media_type": "application/vnd.oci.image.manifest.v1+json",
            "pulp_created": "2020-01-15T12:22:17.724616Z",
            "pulp_href": "/pulp/api/v3/content/container/manifests/14914524-2ee0-4b95-9959-c31a38a376f6/",
            "schema_version": 2
        }
    ]
}
```